src/libraries/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Tests.csproj

```diff
@@ -28,20 +28,17 @@
     <Compile Include="RegexCompilationHelper.cs" />
     <Compile Include="RegexCultureTests.cs" />
     <Compile Include="RegexMatchTimeoutExceptionTests.cs" />
-    <Compile Include="RegexParserTests.cs" />
```

Comment:
Why delete this? Is it because `RegexParseError` is not in Framework and you were changing the exception type in the tests? With this change, won't it mean that we run the parser tests only in netcoreappcurrent? I don't think we want that. Maybe duplicate these tests (and name the new file RegexParserTests.NetCore.cs) and include the new file with your changes in the NetCoreAppCurrent` compile group below?

Reply:
@pgovind, that was the suggestion by @danmosemsft  and @am11. Indeed, only the `RegexParseError` tests are not run in NET48 in this scenario.

I can roll that back, and add back the local linked file to open the private stuff for NET48, but that feels a bit out-of-place, if not only because in part this whole effort was started to get rid of the reflection stuff (that's the only way to get to this info on NET48).

Reply:
Ah, my bad, I made the comment without checking if that file has other tests. It looks like it deals only with parsing. So with this change, we'd just stop testing behavior on Framework. Seems ok to me. I'm not sure how much value testing Framework behavior in this file is giving us anyway.

Reply:
I didn't intend to suggest this - I just supplied the syntax to suppress tests or parts of tests. What about keeping the tests running on .NET Framework, but just not validating the exception type  nor the error (ie., keep the reflection out)

The purpose of running the tests against .NET Framework is to help avoid us accidentally diverging
1. which patterns are valid and which are not
1. matching behavior of valid patterns.

We rely on RegexParserTests.cs to validate the first of these, I think. Of course behavior diverges intentionally, that is why it contains ` [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]` on some of the parser tests, and not others. You have not changed the parsing behavior, so none of the tests should pass or fail that did not before. But on .NET Framework it's fine to just check that there's an ArgumentException thrown (or not) when we throw RegexParseException (or not).

Reply:
Doh - of course the tests need to compile against .NET Framework, and they won't find RegexParseException. Hmm. One approach could be to keep the tests essentially as they are, but change the `private static void Throws(RegexParseError error, Action action)` into a partial method, and have two files: RegexParserTests.netcoreapp.cs, and RegexParserTests.netfx.cs. The former one verifies the RegexParseException, the latter verifies simply there's an ARgumentException, and compile in the appropriate one for the target.

Reply:
Got it. The old code just tested for `ArgumentException` anyway in case of Framework. A simpler solution would be to just put those lines back in, essentially leaving those tests as they were (except for the reflection bit, which isn't necessary anymore, to cross-compile, i can perhaps just use dynamics).

I'll think about this a little.

Reply:
Done, I did it like in prev. comment by @danmosemsft https://github.com/dotnet/runtime/pull/40902/files#r471162733, using partial classes and methods. This proved rather painless. I linked the `RegexParseError` enum so that only one version of the `RegexParserTests.Parse` needs to exist.